src/applications/static-pages/events/components/Results/ResultsWhereContent.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const ResultsWhereContent = ({
+  fieldFacilityLocation,
+  fieldLocationType,
+  locations,
+}) => {
+  function whereContent() {
+    if (fieldLocationType === 'online') return 'This is an online event.';
+    return (
+      <>
+        <a href={fieldFacilityLocation?.entity?.entityUrl.path}>
+          {fieldFacilityLocation?.entity?.title}
+        </a>
+        {locations?.length > 0 && (
+          <div>
+            {locations?.map(location => (
+              <p className="vads-u-margin--0" key={location}>
+                {location}
+              </p>
+            ))}
+          </div>
+        )}
+      </>
+    );
+  }
```

Comment:
I think this could be improved by decoupling this subcomponent a bit. `whereContent` feels like a useful abstraction, but it would be much cleaner to make it a bit more pure by accepting parameters and depending on those parameters for its output. 

You wouldn't need to export this component because it's not needed outside its parent. You don't need to unit test it, arguably, because it's an implementation detail of its parent, which is being tested.

Ex:
```
const WhereContent = (fieldFacilityLocation, fieldLocationType, locations) => {
    if (fieldLocationType === 'online') return 'This is an online event.';
    return (
      <>
        <a href={fieldFacilityLocation?.entity?.entityUrl.path}>
          {fieldFacilityLocation?.entity?.title}
        </a>
        {locations?.length > 0 && (
          <div>
            {locations?.map(location => (
              <p className="vads-u-margin--0" key={location}>
                {location}
              </p>
            ))}
          </div>
        )}
      </>
   );
}
```